### PR TITLE
Fixes 1218: Welcome Card does not display on non-Windows environments

### LINKED ISSUE #1218: Basic bot on mac throws error in BasicBot.cs
## Sample information
1. Sample type: web app bot, 4.x sdk, c#
2. Sample language: dotnetcore
3. Sample name: basic bot

## Describe the bug

Issue #1 - Emulator 4 says it is connected but doesn't get past "couldn't send" regardless if I use development or production endpoint. Verified this also happens on Win10. 

Issue #2 - Emulator 3 throws an error when running against local/development server. Verified this is a mac-only issue. 

## To Reproduce
In azure portal, created a new web app bot, SDK 4.x, c#, basic. 
Downloaded zip file to local mac environment
Unzipped files
Opened solution with mac Visual Studio
Followed readme.md instructions in base of project (adding to appsettings.json file, and using emulator)

`var adaptiveCard = File.ReadAllText(@".\Dialogs\Welcome\Resources\welcomeCard.json");`

The full exception looks like the code isn't dealing with pathing correctly. 

## Expected behavior
Expect sample to work with both emulators and have the exact same behavior as the Portal's "Test in Web Chat" -- which works and shows the cards. 

## Screenshots

snip20190208_9
snip20190208_8
snip20190208_7
snip20190208_6

## Additional context

[bug]

### PR DESCRIPTION
**Pull request by sgellock**
Fixes #1218  

## Proposed Changes
- use Path.Combine to build file path to welcomeCard.json

## Testing
Build and run on macOS
Should see a welcome card as soon as Emulator connects to the Bot

### CODE REVIEW

File: samples/csharp_dotnetcore/13.basic-bot/BasicBot.cs [hunk 1]
```diff
@@ -210,7 +210,9 @@ private Activity CreateResponse(Activity activity, Attachment attachment)
         // Load attachment from file.
         private Attachment CreateAdaptiveCardAttachment()
         {
-            var adaptiveCard = File.ReadAllText(@".\Dialogs\Welcome\Resources\welcomeCard.json");
+            string[] paths = {".", "Dialogs", "Welcome", "Resources", "welcomeCard.json"};
```
Comment (EricDahlvang) [inline samples/csharp_dotnetcore/13.basic-bot/BasicBot.cs]: @sgellock  What is this?  A comment explaining why this is necessary would be good here. (this is weird)

**Comment by sgellock** [comment]
You simply can’t use file paths because the world run on more than just Windows these days.

On Feb 11, 2019, at 8:25 PM, Eric Dahlvang <notifications@github.com<mailto:notifications@github.com>> wrote:


@EricDahlvang commented on this pull request.

________________________________

In samples/csharp_dotnetcore/13.basic-bot/BasicBot.cs<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2FMicrosoft%2FBotBuilder-Samples%2Fpull%2F1223%23discussion_r255798810&data=02%7C01%7Cscott.gellock%40microsoft.com%7C5f09d277a5ea45b68f4308d690a22ba5%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636855423524313753&sdata=1Dcyl1%2BALQBLVLshCer2dq4OsdBwolhF8I5gpVd6jz8%3D&reserved=0>:

> @@ -210,7 +210,9 @@ private Activity CreateResponse(Activity activity, Attachment attachment)
         // Load attachment from file.
         private Attachment CreateAdaptiveCardAttachment()
         {
-            var adaptiveCard = File.ReadAllText(@".\Dialogs\Welcome\Resources\welcomeCard.json");
+            string[] paths = {".", "Dialogs", "Welcome", "Resources", "welcomeCard.json"};


@sgellock<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fsgellock&data=02%7C01%7Cscott.gellock%40microsoft.com%7C5f09d277a5ea45b68f4308d690a22ba5%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636855423524323749&sdata=xM8kaRDcKWLcZzn33wTOoY6gVhqFE%2FLWaGmSQBhOWWg%3D&reserved=0> What is this? A comment explaining why this is necessary would be good here. (this is weird)

**Comment by EricDahlvang** [comment]
Windows + Azure.

**Comment by sgellock** [comment]
I don't understand your comment.  You can't simply hard code file paths in source code that is expected to run on OS's other than windows.  e.g. macOS for local development, Linux distros for deployment.  The code as is simply doesn't run on non-windows OSs.  what additional information do you need to understand the code change?

**Comment by EricDahlvang** [comment]
Sorry, my 'Windows + Azure' comment was facetious.

In seriousness, I've just never seen the syntax here. Concatenating folder names to retrieve a file at run-time.  Maybe I'm just ignorant of this method of cross-platform path resolution method and it will be obvious to more informed developers.

**Comment by sgellock** [comment]
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.combine?view=netframework-4.7.2

**Comment by EricDahlvang** [comment]
Thanks Scott.  Apparently, I'm not keeping up well enough with the cross-platform world.  :/

Here's a PR for some of the other samples:
@sgellock https://github.com/Microsoft/BotBuilder-Samples/pull/1224

**Pull request merged**